src/Keboola/S3Extractor/Extractor.php

```diff
@@ -55,24 +57,13 @@ public function __construct(Config $config, array $state = [], LoggerInterface $
      * @return array
      * @throws UserException
      */
-    public function extract($outputPath)
+    public function extract($outputPath): array
     {
-        $client = new S3MultiRegionClient([
-            'version' => '2006-03-01',
-            'credentials' => [
-                'key' => $this->config->getAccessKeyId(),
-                'secret' => $this->config->getSecretAccessKey(),
-            ],
-        ]);
-        $region = $client->getBucketLocation(["Bucket" => $this->config->getBucket()])->get('LocationConstraint');
-        $client = new S3Client([
-            'region' => $region,
-            'version' => '2006-03-01',
-            'credentials' => [
-                'key' => $this->config->getAccessKeyId(),
-                'secret' => $this->config->getSecretAccessKey(),
-            ],
-        ]);
+        if ($this->config->getLoginType() === 'role') {
```

Comment:
To `'role'` a `'credentials'` sa v kode opakuje, .... asi by to bolo dobre mat ako konstantu, napr. v `ConfigDefinition` triede ... sa sa to potom lahko v IDE vyhladat, kde sa to vsade pouziva.

Reply:
hotovo